AWS Synthetics Canary Missing support for Environment Variables

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

At present, it would appear the AWS Provider does not yet support the ability to set environment variables for `aws_synthetics_canary` resources. As they share some similarities, I was expecting canary resources to support environment variables in a much similar fashion to Lambda Functions.

It would appear that in the original PR to add canary support, there was a attempt to bring in support for environment variables which was subsequently reverted, presumably due to some kind of bug or limitation somewhere. If this was indeed the case, then it would be nice to have this feature included again whenever possible.

### New or Affected Resource(s)

* aws_synthetics_canary

### Potential Terraform Configuration

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
resource "aws_synthetics_canary" "some_canary_test" {
  name                 = "some_canary_test"
  artifact_s3_location = aws_s3_bucket.some_canary_test_bucket.id
  execution_role_arn   = aws_iam_role.some_canary_test_role.arn
  handler              = "test.handler"

  zip_file             = "placeholder_smoke_test.zip"

  runtime_version      = "syn-nodejs-puppeteer-3.1"

  schedule {
    expression = "rate(60 minute)"
  }

  tags = {
    blueprint = "canaryRecorder"
  }

  # This is currently not supported.
  environment {
    variables = {
      API_TOKEN = "some_token_here"
    }
  }
}
```

### References

* #13140

My code is very similar and it fails with this error

ERROR: Canary error: 
Error: Cannot find module '/opt/nodejs/node_modules/exports'Require stack:- /var/task/index.js- /var/runtime/UserFunction.js- /var/runtime/index.js

--- a/.changelog/23574.txt
+++ b/.changelog/23574.txt
@@ -1,3 +1,3 @@
 ```release-note:enhancement
-resource/aws_synthetics_canary: Add optional `environment variables` to `run_config`.
+resource/aws_synthetics_canary: Add optional `environment_variables` to `run_config`.
 ```

--- a/.changelog/23574.txt
+++ b/.changelog/23574.txt
@@ -1,0 +1,3 @@
+```release-note:enhancement
+resource/aws_synthetics_canary: Add optional `environment variables` to `run_config`.
+```

--- a/internal/service/synthetics/canary.go
+++ b/internal/service/synthetics/canary.go
@@ -114,6 +114,11 @@ func ResourceCanary() *schema.Resource {
 							Type:     schema.TypeBool,
 							Optional: true,
 						},
+						"environment_variables": {
+							Type:     schema.TypeMap,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
 						"memory_in_mb": {
 							Type:     schema.TypeInt,
 							Optional: true,
@@ -385,7 +390,13 @@ func resourceCanaryRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error setting vpc config: %w", err)
 	}
 
-	if err := d.Set("run_config", flattenCanaryRunConfig(canary.RunConfig)); err != nil {
+	// get environment variables since they are not available on describe
+	runConfig := &synthetics.CanaryRunConfigInput{}
+	if v, ok := d.GetOk("run_config"); ok {
+		runConfig = expandCanaryRunConfig(v.([]interface{}))
+	}
+
+	if err := d.Set("run_config", flattenCanaryRunConfig(canary.RunConfig, runConfig.EnvironmentVariables)); err != nil {
 		return fmt.Errorf("error setting run config: %w", err)
 	}
 
@@ -703,18 +714,23 @@ func expandCanaryRunConfig(l []interface{}) *synthetics.CanaryRunConfigInput {
 		codeConfig.ActiveTracing = aws.Bool(v)
 	}
 
+	if v, ok := m["enviroment_variables"].(map[string]string); ok {
+		codeConfig.EnvironmentVariables = aws.StringMap(v)
+	}
+
 	return codeConfig
 }
 
-func flattenCanaryRunConfig(canaryCodeOut *synthetics.CanaryRunConfigOutput) []interface{} {
+func flattenCanaryRunConfig(canaryCodeOut *synthetics.CanaryRunConfigOutput, envVars map[string]*string) []interface{} {
 	if canaryCodeOut == nil {
 		return []interface{}{}
 	}
 
 	m := map[string]interface{}{
-		"timeout_in_seconds": aws.Int64Value(canaryCodeOut.TimeoutInSeconds),
-		"memory_in_mb":       aws.Int64Value(canaryCodeOut.MemoryInMB),
-		"active_tracing":     aws.BoolValue(canaryCodeOut.ActiveTracing),
+		"timeout_in_seconds":    aws.Int64Value(canaryCodeOut.TimeoutInSeconds),
+		"memory_in_mb":          aws.Int64Value(canaryCodeOut.MemoryInMB),
+		"active_tracing":        aws.BoolValue(canaryCodeOut.ActiveTracing),
+		"environment_variables": aws.StringValueMap(envVars),
 	}
 
 	return []interface{}{m}

--- a/internal/service/synthetics/canary.go
+++ b/internal/service/synthetics/canary.go
@@ -391,12 +391,15 @@ func resourceCanaryRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	// get environment variables since they are not available on describe
-	runConfig := &synthetics.CanaryRunConfigInput{}
+	envVars := make(map[string]*string)
 	if v, ok := d.GetOk("run_config"); ok {
-		runConfig = expandCanaryRunConfig(v.([]interface{}))
+		runConfig := expandCanaryRunConfig(v.([]interface{}))
+		for k, v := range runConfig.EnvironmentVariables {
+			envVars[k] = v
+		}
 	}
 
-	if err := d.Set("run_config", flattenCanaryRunConfig(canary.RunConfig, runConfig.EnvironmentVariables)); err != nil {
+	if err := d.Set("run_config", flattenCanaryRunConfig(canary.RunConfig, envVars)); err != nil {
 		return fmt.Errorf("error setting run config: %w", err)
 	}
 
@@ -714,8 +717,12 @@ func expandCanaryRunConfig(l []interface{}) *synthetics.CanaryRunConfigInput {
 		codeConfig.ActiveTracing = aws.Bool(v)
 	}
 
-	if v, ok := m["enviroment_variables"].(map[string]string); ok {
-		codeConfig.EnvironmentVariables = aws.StringMap(v)
+	if vars, ok := m["environment_variables"].(map[string]interface{}); ok {
+		ev := make(map[string]string)
+		for k, v := range vars {
+			ev[k] = v.(string)
+		}
+		codeConfig.EnvironmentVariables = aws.StringMap(ev)
 	}
 
 	return codeConfig
@@ -727,10 +734,13 @@ func flattenCanaryRunConfig(canaryCodeOut *synthetics.CanaryRunConfigOutput, env
 	}
 
 	m := map[string]interface{}{
-		"timeout_in_seconds":    aws.Int64Value(canaryCodeOut.TimeoutInSeconds),
-		"memory_in_mb":          aws.Int64Value(canaryCodeOut.MemoryInMB),
-		"active_tracing":        aws.BoolValue(canaryCodeOut.ActiveTracing),
-		"environment_variables": aws.StringValueMap(envVars),
+		"timeout_in_seconds": aws.Int64Value(canaryCodeOut.TimeoutInSeconds),
+		"memory_in_mb":       aws.Int64Value(canaryCodeOut.MemoryInMB),
+		"active_tracing":     aws.BoolValue(canaryCodeOut.ActiveTracing),
+	}
+
+	if v := envVars; v != nil {
+		m["environment_variables"] = aws.StringValueMap(envVars)
 	}
 
 	return []interface{}{m}

--- a/internal/service/synthetics/canary.go
+++ b/internal/service/synthetics/canary.go
@@ -390,16 +390,13 @@ func resourceCanaryRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error setting vpc config: %w", err)
 	}
 
-	// get environment variables since they are not available on describe
-	envVars := make(map[string]*string)
+	// get environment variables since they a
+	runConfig := &synthetics.CanaryRunConfigInput{}
 	if v, ok := d.GetOk("run_config"); ok {
-		runConfig := expandCanaryRunConfig(v.([]interface{}))
-		for k, v := range runConfig.EnvironmentVariables {
-			envVars[k] = v
-		}
+		runConfig = expandCanaryRunConfig(v.([]interface{}))
 	}
 
-	if err := d.Set("run_config", flattenCanaryRunConfig(canary.RunConfig, envVars)); err != nil {
+	if err := d.Set("run_config", flattenCanaryRunConfig(canary.RunConfig, runConfig.EnvironmentVariables)); err != nil {
 		return fmt.Errorf("error setting run config: %w", err)
 	}
 

--- a/internal/service/synthetics/canary.go
+++ b/internal/service/synthetics/canary.go
@@ -390,7 +390,6 @@ func resourceCanaryRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error setting vpc config: %w", err)
 	}
 
-	// get environment variables since they a
 	runConfig := &synthetics.CanaryRunConfigInput{}
 	if v, ok := d.GetOk("run_config"); ok {
 		runConfig = expandCanaryRunConfig(v.([]interface{}))
@@ -715,7 +714,7 @@ func expandCanaryRunConfig(l []interface{}) *synthetics.CanaryRunConfigInput {
 	}
 
 	if vars, ok := m["environment_variables"].(map[string]interface{}); ok && len(vars) > 0 {
-		codeConfig.EnvironmentVariables = flex.ExpandStringMap(ev)
+		codeConfig.EnvironmentVariables = flex.ExpandStringMap(vars)
 	}
 
 	return codeConfig

--- a/internal/service/synthetics/canary.go
+++ b/internal/service/synthetics/canary.go
@@ -736,7 +736,7 @@ func flattenCanaryRunConfig(canaryCodeOut *synthetics.CanaryRunConfigOutput, env
 		"active_tracing":     aws.BoolValue(canaryCodeOut.ActiveTracing),
 	}
 
-	if v := envVars; v != nil {
+	if envVars != nil {
 		m["environment_variables"] = aws.StringValueMap(envVars)
 	}
 

--- a/internal/service/synthetics/canary.go
+++ b/internal/service/synthetics/canary.go
@@ -714,12 +714,8 @@ func expandCanaryRunConfig(l []interface{}) *synthetics.CanaryRunConfigInput {
 		codeConfig.ActiveTracing = aws.Bool(v)
 	}
 
-	if vars, ok := m["environment_variables"].(map[string]interface{}); ok {
-		ev := make(map[string]string)
-		for k, v := range vars {
-			ev[k] = v.(string)
-		}
-		codeConfig.EnvironmentVariables = aws.StringMap(ev)
+	if vars, ok := m["environment_variables"].(map[string]interface{}); ok && len(vars) > 0 {
+		codeConfig.EnvironmentVariables = flex.ExpandStringMap(ev)
 	}
 
 	return codeConfig

--- a/internal/service/synthetics/canary_test.go
+++ b/internal/service/synthetics/canary_test.go
@@ -319,6 +319,7 @@ func TestAccSyntheticsCanary_run(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "name", rName),
 					resource.TestCheckResourceAttr(resourceName, "run_config.0.memory_in_mb", "1000"),
 					resource.TestCheckResourceAttr(resourceName, "run_config.0.timeout_in_seconds", "60"),
+					resource.TestCheckResourceAttr(resourceName, "run_config.0.environment_variables.foo", "bar"),
 				),
 			},
 			{
@@ -761,7 +762,10 @@ resource "aws_synthetics_canary" "test" {
   }
 
   run_config {
-    timeout_in_seconds = 60
+    timeout_in_seconds    = 60
+	environment_variables = {
+		foo = "bar"
+	}
   }
 
   depends_on = [aws_iam_role.test, aws_iam_role_policy.test]

--- a/internal/service/synthetics/canary_test.go
+++ b/internal/service/synthetics/canary_test.go
@@ -319,14 +319,14 @@ func TestAccSyntheticsCanary_run(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "name", rName),
 					resource.TestCheckResourceAttr(resourceName, "run_config.0.memory_in_mb", "1000"),
 					resource.TestCheckResourceAttr(resourceName, "run_config.0.timeout_in_seconds", "60"),
-					resource.TestCheckResourceAttr(resourceName, "run_config.0.environment_variables.foo", "bar"),
+					resource.TestCheckResourceAttr(resourceName, "run_config.0.environment_variables.test1", "result1"),
 				),
 			},
 			{
 				ResourceName:            resourceName,
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"zip_file", "start_canary"},
+				ImportStateVerifyIgnore: []string{"zip_file", "start_canary", "run_config.0.environment_variables"},
 			},
 			{
 				Config: testAccCanaryRun2Config(rName),
@@ -334,6 +334,7 @@ func TestAccSyntheticsCanary_run(t *testing.T) {
 					testAccCheckCanaryExists(resourceName, &conf),
 					resource.TestCheckResourceAttr(resourceName, "run_config.0.memory_in_mb", "960"),
 					resource.TestCheckResourceAttr(resourceName, "run_config.0.timeout_in_seconds", "120"),
+					resource.TestCheckResourceAttr(resourceName, "run_config.0.environment_variables.test2", "result2"),
 				),
 			},
 			{
@@ -764,7 +765,7 @@ resource "aws_synthetics_canary" "test" {
   run_config {
     timeout_in_seconds    = 60
 	environment_variables = {
-		foo = "bar"
+		test1 = "result1"
 	}
   }
 
@@ -790,6 +791,10 @@ resource "aws_synthetics_canary" "test" {
   run_config {
     timeout_in_seconds = 120
     memory_in_mb       = 960
+	environment_variables = {
+		test1  = "result1"
+		test2 = "result2"
+	}
   }
 
   depends_on = [aws_iam_role.test, aws_iam_role_policy.test]

--- a/internal/service/synthetics/canary_test.go
+++ b/internal/service/synthetics/canary_test.go
@@ -764,9 +764,8 @@ resource "aws_synthetics_canary" "test" {
 
   run_config {
     timeout_in_seconds = 60
-	
     environment_variables = {
-        test1 = "result1"
+      test1 = "result1"
     }
   }
 
@@ -792,10 +791,9 @@ resource "aws_synthetics_canary" "test" {
   run_config {
     timeout_in_seconds = 120
     memory_in_mb       = 960
-	
     environment_variables = {
-        test1  = "result1"
-        test2  = "result2"
+      test1 = "result1"
+      test2 = "result2"
     }
   }
 

--- a/internal/service/synthetics/canary_test.go
+++ b/internal/service/synthetics/canary_test.go
@@ -789,11 +789,11 @@ resource "aws_synthetics_canary" "test" {
   }
 
   run_config {
-    timeout_in_seconds = 120
-    memory_in_mb       = 960
+    timeout_in_seconds    = 120
+    memory_in_mb          = 960
 	environment_variables = {
 		test1  = "result1"
-		test2 = "result2"
+		test2  = "result2"
 	}
   }
 

--- a/internal/service/synthetics/canary_test.go
+++ b/internal/service/synthetics/canary_test.go
@@ -763,10 +763,11 @@ resource "aws_synthetics_canary" "test" {
   }
 
   run_config {
-    timeout_in_seconds    = 60
-	environment_variables = {
-		test1 = "result1"
-	}
+    timeout_in_seconds = 60
+	
+    environment_variables = {
+        test1 = "result1"
+    }
   }
 
   depends_on = [aws_iam_role.test, aws_iam_role_policy.test]
@@ -789,12 +790,13 @@ resource "aws_synthetics_canary" "test" {
   }
 
   run_config {
-    timeout_in_seconds    = 120
-    memory_in_mb          = 960
-	environment_variables = {
-		test1  = "result1"
-		test2  = "result2"
-	}
+    timeout_in_seconds = 120
+    memory_in_mb       = 960
+	
+    environment_variables = {
+        test1  = "result1"
+        test2  = "result2"
+    }
   }
 
   depends_on = [aws_iam_role.test, aws_iam_role_policy.test]

--- a/internal/service/synthetics/canary_test.go
+++ b/internal/service/synthetics/canary_test.go
@@ -391,6 +391,49 @@ func TestAccSyntheticsCanary_runTracing(t *testing.T) {
 	})
 }
 
+func TestAccSyntheticsCanary_runEnvironmentVariables(t *testing.T) {
+	var conf synthetics.Canary
+	rName := fmt.Sprintf("tf-acc-test-%s", sdkacctest.RandString(8))
+	resourceName := "aws_synthetics_canary.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, synthetics.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckCanaryDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCanaryRunEnvVariables1Config(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckCanaryExists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "run_config.0.environment_variables.test1", "result1"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"zip_file", "start_canary", "run_config.0.environment_variables"},
+			},
+			{
+				Config: testAccCanaryRunEnvVariables2Config(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckCanaryExists(resourceName, &conf),
+					resource.TestCheckResourceAttr(resourceName, "run_config.0.environment_variables.test1", "result1"),
+					resource.TestCheckResourceAttr(resourceName, "run_config.0.environment_variables.test2", "result2"),
+				),
+			},
+			{
+				Config: testAccCanaryRunEnvVariables3Config(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckCanaryExists(resourceName, &conf),
+					resource.TestCheckNoResourceAttr(resourceName, "run_config.0.environment_variables"),
+				),
+			},
+		},
+	})
+}
+
 func TestAccSyntheticsCanary_vpc(t *testing.T) {
 	if testing.Short() {
 		t.Skip("skipping long-running test in short mode")
@@ -764,9 +807,6 @@ resource "aws_synthetics_canary" "test" {
 
   run_config {
     timeout_in_seconds = 60
-    environment_variables = {
-      test1 = "result1"
-    }
   }
 
   depends_on = [aws_iam_role.test, aws_iam_role_policy.test]
@@ -791,10 +831,6 @@ resource "aws_synthetics_canary" "test" {
   run_config {
     timeout_in_seconds = 120
     memory_in_mb       = 960
-    environment_variables = {
-      test1 = "result1"
-      test2 = "result2"
-    }
   }
 
   depends_on = [aws_iam_role.test, aws_iam_role_policy.test]
@@ -824,6 +860,76 @@ resource "aws_synthetics_canary" "test" {
   depends_on = [aws_iam_role.test, aws_iam_role_policy.test]
 }
 `, rName, tracing))
+}
+
+func testAccCanaryRunEnvVariables1Config(rName string) string {
+	return acctest.ConfigCompose(testAccCanaryBaseConfig(rName), fmt.Sprintf(`
+resource "aws_synthetics_canary" "test" {
+  name                 = %[1]q
+  artifact_s3_location = "s3://${aws_s3_bucket.test.bucket}/"
+  execution_role_arn   = aws_iam_role.test.arn
+  handler              = "exports.handler"
+  zip_file             = "test-fixtures/lambdatest.zip"
+  runtime_version      = "syn-nodejs-puppeteer-3.2"
+
+  schedule {
+    expression = "rate(0 minute)"
+  }
+
+  run_config {
+    environment_variables = {
+	  test1 = "result1"
+	}
+  }
+
+  depends_on = [aws_iam_role.test, aws_iam_role_policy.test]
+}
+`, rName))
+}
+
+func testAccCanaryRunEnvVariables2Config(rName string) string {
+	return acctest.ConfigCompose(testAccCanaryBaseConfig(rName), fmt.Sprintf(`
+resource "aws_synthetics_canary" "test" {
+  name                 = %[1]q
+  artifact_s3_location = "s3://${aws_s3_bucket.test.bucket}/"
+  execution_role_arn   = aws_iam_role.test.arn
+  handler              = "exports.handler"
+  zip_file             = "test-fixtures/lambdatest.zip"
+  runtime_version      = "syn-nodejs-puppeteer-3.2"
+
+  schedule {
+    expression = "rate(0 minute)"
+  }
+
+  run_config {
+    environment_variables = {
+	  test1 = "result1"
+      test2 = "result2"
+	}
+  }
+
+  depends_on = [aws_iam_role.test, aws_iam_role_policy.test]
+}
+`, rName))
+}
+
+func testAccCanaryRunEnvVariables3Config(rName string) string {
+	return acctest.ConfigCompose(testAccCanaryBaseConfig(rName), fmt.Sprintf(`
+resource "aws_synthetics_canary" "test" {
+  name                 = %[1]q
+  artifact_s3_location = "s3://${aws_s3_bucket.test.bucket}/"
+  execution_role_arn   = aws_iam_role.test.arn
+  handler              = "exports.handler"
+  zip_file             = "test-fixtures/lambdatest.zip"
+  runtime_version      = "syn-nodejs-puppeteer-3.2"
+
+  schedule {
+    expression = "rate(0 minute)"
+  }
+
+  depends_on = [aws_iam_role.test, aws_iam_role_policy.test]
+}
+`, rName))
 }
 
 func testAccCanaryBasicConfig(rName string) string {

--- a/internal/service/synthetics/canary_test.go
+++ b/internal/service/synthetics/canary_test.go
@@ -319,14 +319,13 @@ func TestAccSyntheticsCanary_run(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "name", rName),
 					resource.TestCheckResourceAttr(resourceName, "run_config.0.memory_in_mb", "1000"),
 					resource.TestCheckResourceAttr(resourceName, "run_config.0.timeout_in_seconds", "60"),
-					resource.TestCheckResourceAttr(resourceName, "run_config.0.environment_variables.test1", "result1"),
 				),
 			},
 			{
 				ResourceName:            resourceName,
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"zip_file", "start_canary", "run_config.0.environment_variables"},
+				ImportStateVerifyIgnore: []string{"zip_file", "start_canary"},
 			},
 			{
 				Config: testAccCanaryRun2Config(rName),
@@ -334,7 +333,6 @@ func TestAccSyntheticsCanary_run(t *testing.T) {
 					testAccCheckCanaryExists(resourceName, &conf),
 					resource.TestCheckResourceAttr(resourceName, "run_config.0.memory_in_mb", "960"),
 					resource.TestCheckResourceAttr(resourceName, "run_config.0.timeout_in_seconds", "120"),
-					resource.TestCheckResourceAttr(resourceName, "run_config.0.environment_variables.test2", "result2"),
 				),
 			},
 			{
@@ -424,7 +422,7 @@ func TestAccSyntheticsCanary_runEnvironmentVariables(t *testing.T) {
 				),
 			},
 			{
-				Config: testAccCanaryRunEnvVariables3Config(rName),
+				Config: testAccCanaryBasicConfig(rName),
 				Check: resource.ComposeAggregateTestCheckFunc(
 					testAccCheckCanaryExists(resourceName, &conf),
 					resource.TestCheckNoResourceAttr(resourceName, "run_config.0.environment_variables"),
@@ -878,8 +876,8 @@ resource "aws_synthetics_canary" "test" {
 
   run_config {
     environment_variables = {
-	  test1 = "result1"
-	}
+      test1 = "result1"
+    }
   }
 
   depends_on = [aws_iam_role.test, aws_iam_role_policy.test]
@@ -903,28 +901,9 @@ resource "aws_synthetics_canary" "test" {
 
   run_config {
     environment_variables = {
-	  test1 = "result1"
+      test1 = "result1"
       test2 = "result2"
-	}
-  }
-
-  depends_on = [aws_iam_role.test, aws_iam_role_policy.test]
-}
-`, rName))
-}
-
-func testAccCanaryRunEnvVariables3Config(rName string) string {
-	return acctest.ConfigCompose(testAccCanaryBaseConfig(rName), fmt.Sprintf(`
-resource "aws_synthetics_canary" "test" {
-  name                 = %[1]q
-  artifact_s3_location = "s3://${aws_s3_bucket.test.bucket}/"
-  execution_role_arn   = aws_iam_role.test.arn
-  handler              = "exports.handler"
-  zip_file             = "test-fixtures/lambdatest.zip"
-  runtime_version      = "syn-nodejs-puppeteer-3.2"
-
-  schedule {
-    expression = "rate(0 minute)"
+    }
   }
 
   depends_on = [aws_iam_role.test, aws_iam_role_policy.test]

--- a/website/docs/r/synthetics_canary.html.markdown
+++ b/website/docs/r/synthetics_canary.html.markdown
@@ -73,6 +73,7 @@ The following arguments are optional:
 * `timeout_in_seconds` - (Optional) Number of seconds the canary is allowed to run before it must stop. If you omit this field, the frequency of the canary is used, up to a maximum of 840 (14 minutes).
 * `memory_in_mb` - (Optional) Maximum amount of memory available to the canary while it is running, in MB. The value you specify must be a multiple of 64.
 * `active_tracing` - (Optional) Whether this canary is to use active AWS X-Ray tracing when it runs. You can enable active tracing only for canaries that use version syn-nodejs-2.0 or later for their canary runtime.
+* `environment_variables` - (Optional) Map of environment variables that are accessible from the canary during execution. Please see [AWS Docs](https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime) for variables reserved for Lambda.
 
 ### vpc_config
 

